File: src/game/model/position.ts
import { Direction } from './direction.ts';

export class Position {
  constructor(public x: number, public y: number) {}
  getDirectionTo(target: Position) { return new Direction(target.x - this.x, target.y - this.y); }
  distanceTo(target: Position) { return Math.hypot(this.x - target.x, this.y - target.y); }
}

File: app/src/main/java/space/model/Position.java
package space.model;




public class Position  {
	
	public float x;
	public float y;

	public Position(float x, float y) {
		this.x = x;
		this.y = y;
	}

	public Direction getDirectionTo(Position aimAt) {
		
		return new Direction (aimAt.x - x, aimAt.y - y) ;
	}

	public float distanceTo(Position aimAt) {
		return (float) Math.sqrt((x - aimAt.x) * (x - aimAt.x) + (y - aimAt.y) * (y - aimAt.y));
	}

}
